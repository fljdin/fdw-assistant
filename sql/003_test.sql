
SET tools.targets = 'public.t1';

-- Should return a single job for public.t1
SELECT target, statement FROM run();

SELECT run_id, job_id, config_id, lastseq, rows, state
  FROM job WHERE run_id = 3;

RESET tools.targets;

-- Should return all jobs
SELECT target, statement FROM run();

SELECT run_id, job_id, config_id, lastseq, rows, state
  FROM job WHERE run_id = 4;
