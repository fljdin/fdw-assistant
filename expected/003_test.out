-- Should return a single job for public.t1
SELECT target, statement FROM plan('{public.t1}');
  target   |     statement     
-----------+-------------------
 public.t1 | CALL fdw.copy(7);
(1 row)

SELECT stage_id, job_id, config_id, lastseq, rows, state
  FROM job WHERE stage_id = 3;
 stage_id | job_id | config_id | lastseq | rows |  state  
----------+--------+-----------+---------+------+---------
        3 |      7 |         1 |       0 |    0 | pending
(1 row)

-- Should return all jobs
SELECT target, statement FROM plan();
  target   |     statement      
-----------+--------------------
 public.t2 | CALL fdw.copy(9);
 public.t2 | CALL fdw.copy(10);
 public.t1 | CALL fdw.copy(8);
(3 rows)

SELECT stage_id, job_id, config_id, lastseq, rows, state
  FROM job WHERE stage_id = 4;
 stage_id | job_id | config_id | lastseq | rows |  state  
----------+--------+-----------+---------+------+---------
        4 |      8 |         1 |       0 |    0 | pending
        4 |      9 |         2 |    2000 |    0 | pending
        4 |     10 |         3 |    1999 |    0 | pending
(3 rows)

-- Should fail as the table does not exist
SELECT target, statement FROM plan('{public.foo}');
ERROR:  relation "public.foo" does not exist
CONTEXT:  SQL function "plan" statement 1
-- Should return an empty set as the table has no configuration
SELECT target, statement FROM plan('{public.dummy}');
 target | statement 
--------+-----------
(0 rows)

